### PR TITLE
UI: Update some background colors

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 1]
```diff
@@ -96,6 +97,11 @@ class MainActivity : ThemedActivity() {
         SyncMessagesWorker.cancelWork(this)
     }
 
+    override fun onResume() {
+        super.onResume()
+        if (binding.drawerLayout.isOpen) colorSystemBarsWithMenuDrawer()
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:102]: Why from resume ?

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 2]
```diff
@@ -198,4 +205,14 @@ class MainActivity : ThemedActivity() {
             showSnackbar(title, onActionClicked = onActionClicked)
         }
     }
+
+    private fun colorSystemBarsWithMenuDrawer(slideOffset: Float = 1.0f) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:209]: Magic number, extract to companion.

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 3]
```diff
@@ -198,4 +205,14 @@ class MainActivity : ThemedActivity() {
...
+        if (slideOffset == 1.0f) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:210]: Magic number, extract to companion.

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 4]
```diff
@@ -53,14 +53,15 @@ class MainActivity : ThemedActivity() {
 
     private val backgroundColor: Int by lazy { getColor(R.color.backgroundColor) }
     private val backgroundHeaderColor: Int by lazy { getColor(R.color.backgroundHeaderColor) }
+    private val menuDrawerBackground: Int by lazy { getColor(R.color.menuDrawerBackground) }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:56]: Why not naming it menuDrawerBackgroundColor or backgroundMenuDrawerColor to be consistent ?

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:56]: `backgroundMenuDrawerColor` 👌

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:56]:
```suggestion
    private val backgroundMenuDrawerColor: Int by lazy { getColor(R.color.backgroundMenuDrawerColor) }
```

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:56]: And `R.color.backgroundMenuDrawerColor` need to be renamed accordingly.

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt (hunk 1 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:103]: This only works in onResume, else the drawer is never detected as being "open"

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 5]
```diff
@@ -198,4 +206,19 @@ class MainActivity : ThemedActivity() {
             showSnackbar(title, onActionClicked = onActionClicked)
         }
     }
+
+
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:210]:
```suggestion
```

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**